src/input/code_templates/c/static_code/exi_basetypes.c.jinja

```diff
@@ -97,4 +117,80 @@ int exi_basetypes_convert_64_from_unsigned(exi_unsigned_t* exi_unsigned, uint64_
 
     return EXI_ERROR__NO_ERROR;
 }
+
+int exi_basetypes_convert_from_signed(const exi_signed_t* exi_signed, int32_t* value, size_t max_octets)
+{
+    uint32_t uValue = 0;
+    int res = exi_basetypes_convert_from_unsigned(&exi_signed->data, &uValue, max_octets);
+    *value = (exi_signed->is_negative == 0) ? uValue : -uValue;
+    return res;
+}
+
+int exi_basetypes_convert_64_from_signed(const exi_signed_t* exi_signed, int64_t* value)
+{
+    uint64_t uValue = 0;
+    int res = exi_basetypes_convert_64_from_unsigned(&exi_signed->data, &uValue);
+    *value = (exi_signed->is_negative == 0) ? uValue : -uValue;
+    return res;
+}
+
+int exi_basetypes_convert_bytes_from_unsigned(const exi_unsigned_t* exi_unsigned, uint8_t* data, size_t* data_len, size_t data_size)
+{    
+    const uint8_t* current_octet = exi_unsigned->octets;
+    uint16_t temp = 0;
+    *data_len = 0;
+    size_t total_offset = 0;
+
+    for (size_t n = 0; n < exi_unsigned->octets_count; n++) {
+        temp = temp + ((uint16_t)(*current_octet & EXI_BASETYPES_OCTET_SEQ_VALUE_MASK) << (total_offset));
+        total_offset += 7;
+        if (total_offset >= 8) {
+            if (data_size == *data_len) {
+                return -1;
+            }
+            total_offset -= 8;
+            data[(*data_len)++] = temp & 0xFF;
+            temp >>= 8;
+        }
+        current_octet++;
+    }
+    if (total_offset != 0) {
+        if (data_size == *data_len) {
+            return -1;
+        }
+        data[(*data_len)++] = temp & 0xFF;
+    }
+    return 0;
+}
+
+int exi_basetypes_convert_bytes_to_unsigned(exi_unsigned_t* exi_unsigned, const uint8_t* data, size_t data_len)
```

Comment:
as said in the other place: the object is likely to be larger than a register, so it will be copied on the stack of the function, which may have negative performance implications.

I have not tested this, nor do I have the means to on short notice.